sdk/storage/azure-storage-common/src/shared_key_policy_lite.cpp

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "azure/storage/common/internal/shared_key_policy_lite.hpp"
+
+#include "azure/storage/common/crypt.hpp"
+
+#include <azure/core/http/http.hpp>
+#include <azure/core/internal/strings.hpp>
+
+#include <algorithm>
+
+namespace Azure { namespace Storage { namespace _internal {
+
+  std::string SharedKeyPolicyLite::GetSignature(const Core::Http::Request& request) const
+  {
+    std::string string_to_sign;
+
+    const auto& headers = request.GetHeaders();
+
+    // canonical date header headers
+    const std::string dateHeader = headers.at("x-ms-date");
+    string_to_sign += dateHeader + "\n";
+
+    // If the request URI addresses a component of the resource, append the appropriate query
+    // string. The query string should include the question mark and the comp parameter (for
+    // example, ?comp=metadata). No other parameters should be included on the query string.
+    // https://docs.microsoft.com/en-us/rest/api/storageservices/authorize-with-shared-key#shared-key-lite-and-table-service-format-for-2009-09-19-and-later
+    // canonicalized resource
+    string_to_sign += "/" + m_credential->AccountName + "/" + request.GetUrl().GetPath();
+    auto queryParameters = request.GetUrl().GetQueryParameters();
+    if (queryParameters.count("comp") > 0)
+    {
+      const std::string keyValue = Azure::Core::Url::Encode("comp");
+      auto compValue = queryParameters.at(keyValue);
+      string_to_sign += "?comp=" + Azure::Core::Url::Decode(compValue);
+    }
```

Comment:
we ahve tests for that.

Reply:
Can you please point me to one that is testing this code path, to save some time in the review? It is hard for me to validate correctness here through code inspection.

Reply:
https://github.com/Azure/azure-sdk-for-cpp/pull/5137/files#diff-2e96bb6a731603e3c25c720abffcd7bede63030aabdaee94b32e627265536c5f 

i can only check for the length of the string, i cannot check the string itself as it causes the checks for passwords to trigger because it's a weird value.